heron/instance/src/java/com/twitter/heron/instance/Slave.java

```diff
@@ -179,6 +179,8 @@ public void close() {
     LOG.info("Closing the Slave Thread");
     this.metricsCollector.forceGatherAllMetrics();
     LOG.info("Cleaning up the instance");
-    instance.stop();
+    if (instance != null) {
```

Comment:
I'd make the argument that checking for null before closing a resource if fairly common practice and not in need of an explanation.